openpype/modules/sync_server/sync_server.py

```diff
@@ -189,6 +202,133 @@ def _site_is_working(module, project_name, site_name, site_config):
+        anatomy (Anatomy, optional): Anatomy (Used for optimization).
+            Defaults to None.
+        asset_doc (dict, optional): Asset doc (Used for optimization).
+            Defaults to None.
+
+    Returns:
+        str: New local workfile path.
+    """
+
+    if not anatomy:
+        anatomy = Anatomy(project_name)
+
+    if not asset_doc:
+        asset_doc = get_asset_by_name(project_name, asset_name)
+
+    # Get sync server module
+    sync_server = ModulesManager().modules_by_name.get("sync_server")
+    if not sync_server or not sync_server.enabled:
+        print("Sync server module is disabled or unavailable.")
+        return
+
+    if not workfile_representation:
+        print(
+            "Not published workfile for task '{}' and host '{}'.".format(
+                task_name, host_name
+            )
+        )
+        return
+
+    # If representation isn't available on remote site, then return.
+    if not sync_server.is_representation_on_site(
+        project_name,
+        workfile_representation["_id"],
+        sync_server.get_remote_site(project_name),
+    ):
+        print(
+            "Representation for task '{}' and host '{}'".format(
+                task_name, host_name
+            )
+        )
+        return
+
+    # Get local site
+    local_site_id = get_local_site_id()
+
+    # Add workfile representation to local site
+    representation_ids = {workfile_representation["_id"]}
+    representation_ids.update(
+        get_linked_representation_id(
+            project_name, repre_id=workfile_representation["_id"]
+        )
+    )
+    for repre_id in representation_ids:
+        if not sync_server.is_representation_on_site(project_name, repre_id,
+                                                     local_site_id):
+            sync_server.add_site(
+                project_name,
+                repre_id,
+                local_site_id,
+                force=True,
+                priority=99
+            )
+    sync_server.reset_timer()
+
+    # While representation unavailable locally, wait.
+    while not sync_server.is_representation_on_site(
+        project_name, workfile_representation["_id"], local_site_id,
+        max_retries=max_retries
+    ):
+        sleep(5)
+
+    if not last_published_workfile_path:
+        last_published_workfile_path = get_representation_path_with_anatomy(
+            workfile_representation, anatomy
+        )
+    project_doc = get_project(project_name)
+
+    # Get workfile data
+    workfile_data = get_template_data(
+        project_doc, asset_doc, task_name, host_name
+    )
+
+    extension = last_published_workfile_path.split(".")[-1]
+
+    project_settings = get_project_settings(project_name)
+    template_key = get_workfile_template_key(
+        task_name, host_name, project_name, project_settings
+    )
+
+    workfile_data["version"] = (
+        workfile_representation["context"]["version"] + 1)
```

Comment:
Also when talking about copying, this doesn't look right either. Publish and Workfile may have different versioning (there could be workfile v100 but only published workfile v1.

Reply:
This gets triggered only if there isn't any workfile present at all. It looks only for published.

We are not syncing work workfiles. so even if there would be v100 in studio's `work`, it gets ignored.